pkg/sharing/secret_reconciler.go

```diff
@@ -116,6 +116,19 @@ const (
 
 func (r *SecretReconciler) predictWantToReconcile(secret corev1.Secret) bool {
 	_, found := secret.Annotations[imagePullSecretAnnKey]
+	if found {
+		query := types.NamespacedName{
+			Name: secret.Namespace,
+		}
+		namespace := corev1.Namespace{}
+		err := r.client.Get(context.Background(), query, &namespace)
+		if err != nil {
+			r.log.Error(err, "predictWantToReconcile couldn't find:", "namespace", secret.Namespace)
+			return false
+		}
+		_, excluded := namespace.Annotations["secretgen.carvel.dev/excluded-from-wildcard-matching"]
+		found = !excluded
```

Comment:
predictWantToReconcile is not a "chokepoint" for whether to reconcile something or not -- Reconcile method in theory could be called at the start of the controller on all secrets.

the main chokepoint is https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/pkg/sharing/secret_exports.go#L96 where we determine which secrets should be imported (this also affects SecretImport controller). if destination namespace is marked with above annotation, _and_ if SecretExport is matching it because of "*", then we should not include this secret. SecretExports with explicit namespace name are still allowed.